lib/snippets/kubernetes-deploy

```diff
@@ -66,99 +200,455 @@ class KubernetesDeploy
       errors << "#{@template_path} doesn't contain valid templates (postfix .yml or .yml.erb)"
     end
 
-    if @namespace.nil? || @namespace.empty?
+    if @namespace.blank?
       errors << "Namespace must be specified"
     end
 
-    if @context.nil? || @context.empty?
+    if @context.blank?
       errors << "Context must be specified"
     end
 
     raise FatalDeploymentError, "Configuration invalid: #{errors.join(", ")}" unless errors.empty?
-    logger.info("All required parameters and files are present")
-  end
-
-  def apply_all_templates
-    found = false
-    Dir.foreach(@template_path) do |file|
-      file_path = "#{@template_path}/#{file}"
-      case File.extname(file)
-      when '.yml'
-        found ||= true
-        apply_template(file_path)
-      when '.erb'
-        found ||= true
-        render_and_apply_template(file_path)
-      end
-    end
-    raise FatalDeploymentError, "No templates found in #{@template_path}" unless found
+    KubernetesDeploy.logger.info("All required parameters and files are present")
   end
 
-  def render_and_apply_template(file_path)
-    erb_template = ERB.new(File.read(file_path))
-    erb_binding = binding
-    template_variables.each do |var_name, value|
-      erb_binding.local_variable_set(var_name, value)
+  def deploy_resources(resources, prune: false)
+    command = ["apply", "--namespace=#{@namespace}"]
+    KubernetesDeploy.logger.info("Deploying resources:")
+
+    resources.each do |r|
+      KubernetesDeploy.logger.info("- #{r.id}")
+      command.push("-f", r.file.path)
+      r.deploy_started = Time.now.utc
     end
-    content = erb_template.result(erb_binding)
 
-    f = Tempfile.new(['kube_template', '.yml'])
-    f.write(content)
-    f.close
-    apply_template(f.path, original_path: file_path)
-  end
+    if prune
+      command.push("--prune", "--all")
+      PRUNE_WHITELIST.each { |type| command.push("--prune-whitelist=#{type}") }
```

Comment:
Yes, definitely. In fact, this seems to be the only way to do it... I initially got frustrated trying to do things like `=[list]` `=thing1,thing2`. Turns out there's an example on the PR that introduced the option: https://github.com/kubernetes/kubernetes/pull/35220